extensions/default/src/DicomWebDataSource/wado/retrieveMetadataLoaderAsync.js

```diff
@@ -92,6 +92,21 @@ export default class RetrieveMetadataLoaderAsync extends RetrieveMetadataLoader
 
     const promises = [];
 
+    // if a large study, fetch first a small group of series to run hanging protocol
+    if (preLoadData.length > this.largeStudySeriesCountThreshold) {
+      const firstGroup = [];
+      while (seriesAsyncLoader.hasNext()) {
+        const promise = seriesAsyncLoader.next();
+        promises.push(promise);
+        firstGroup.push(promise);
+        if (firstGroup.length >= this.initialSeriesFetchSize) {
+          break;
+        }
+      }
+      // wait for all promises in the first group to be settled
+      await Promise.allSettled(firstGroup);
```

Comment:
This doesn't quite feel like it should be here - this method is supposed to return immediately, after having initiated the various requests, just providing the set of promises.  What you want is to be able to group the requests so that there are no more than N of them concurrently, but let them proceed as fast as possible, just allowing that many at once.
Then, once the first set of instances is settled, you can return, but you might well get many more than just than just N of them.  Probably return 2 promises, one of which is the initial group list - which you always return regardless of size, and the second of which is the all settled on everything promise.  Code is a bit of a pain for this as you really do care that it is the first N of them, and not just a random set.

Reply:
Thinking about how to implement that, what you want is a set of promises which don't initiate the actual retrieve, but have an initiate retrieve call as part of the promise.  The retrieve initiate call does:
1. Setup the http retrieve
2. When the http retrieve completes, find the next promise in order to retrieve, and call that initiate method
3. Complete the promise
Then, the retrieve metadata loader async knows nothing about retrieve a first data set, it simply always knows to retrieve the metadata in teh sorted order, working in groups of N retrieves at a time, where N is defined by the data source.

Reply:
Thought about this more last night, what you want to do is ALWAYS use an allSettled - but the choice of which promises are in the allSettled simply depends on the configured number of items to include that are required to run the hanging protocol.  Then, when the allSettled completes, you apply the hanging protocol.
The code described in the async loader ensures that the instances are loaded in order.  It is also possible to add another initiate if specific instances are required in other orders.

Reply:
I feel like creating this will make a lot of internal changes that I dont know this PR should cover, because all the code expects the promises array has one promise per serie